src/components/UsernameEntry/actions.js

```diff
@@ -54,17 +56,24 @@ export const sendSlackMessage = userid => async (dispatch, getState) => {
 
   try {
     const result = await retry(
-      () =>
-        fetch(`https://slack.com/api/chat.postMessage?token=${token}&
-      channel=${userid}&icon_url=https://honesty.store/assets/android/icon@MDPI.png&username=honesty.store&
-      text=${`Click to purchase your ${itemName}: https://honesty.store/item/${actualItemID}`}`),
+      () => attemptSendSlackMessage(userid, itemName, actualItemID),
       () => dispatch(setSendMessageError(true))
-    ).then(response => {
-      dispatch(setSendMessageError(false));
-      return response.json();
-    });
+    );
+    dispatch(setSendMessageError(false));
+
     return result.ok;
   } catch (error) {
     return false;
   }
 };
+
+const attemptSendSlackMessage = async (userid, itemName, actualItemID) => {
+  const send = firebase.functions().httpsCallable('sendSlackMessage');
+  const result = await send({userid, itemName, actualItemID}).then(response => {
+    return response.data;
+  });
+  if (!result.ok) {
+    throw Error('Unknown Error');
```

Comment:
Could we have a more informative error message? Such as "A network error occured".